homeassistant/util/dt.py

```diff
@@ -4,20 +4,16 @@
 from contextlib import suppress
 import datetime as dt
 import re
-from typing import Any, cast
+from typing import Any
 
 import ciso8601
-import pytz
-import pytz.exceptions as pytzexceptions
-import pytz.tzinfo as pytzinfo
+from dateutil import tz
 
 from homeassistant.const import MATCH_ALL
 
 DATE_STR_FORMAT = "%Y-%m-%d"
-NATIVE_UTC = dt.timezone.utc
-UTC = pytz.utc
-DEFAULT_TIME_ZONE: dt.tzinfo = pytz.utc
-
+UTC = dt.timezone.utc
+DEFAULT_TIME_ZONE: dt.tzinfo = dt.timezone.utc
 
 # Copyright (c) Django Software Foundation and individual contributors.
```

Comment:
Thinking out loud here, and probably for a future PR, but we can drop this regex and `parse_datetime` helper function in favor of [`dateutil.parser.isoparse`](https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse). This is also recommended by Python core: https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat

This function was adopted from Django when it wasn't in core yet.